### PR TITLE
allow target subsets too

### PR DESCRIPTION
**Pull request by sprivite**
Implements issue #19

### CODE REVIEW

File: pybalance/utils/balance_calculators.py [hunk 1]
```diff
@@ -248,39 +248,56 @@ def _to_list(self, candidate_populations):
 
     @reshape_output
     def distance(
-        self, candidate_populations: Union[pd.DataFrame, torch.Tensor, np.ndarray]
+        self,
```
Comment (a-hartens) [inline pybalance/utils/balance_calculators.py:251]: seemingly inconsistent use of 'pool' to 'full population data' name change seen in _fetch_features?

File: pybalance/utils/balance_calculators.py [hunk 2]
```diff
@@ -216,17 +216,17 @@ def _preprocess(self, data: Union[pd.DataFrame, MatchingData]):
         )
         return data
 
-    def _fetch_pool_features(self, candidate_populations):
-        if isinstance(candidate_populations, pd.DataFrame):
-            features = self._preprocess(candidate_populations)
+    def _fetch_features(self, subset_populations, full_population_data):
```
Comment (a-hartens) [inline pybalance/utils/balance_calculators.py:219]: i'm seeing that later in the file you are still referring to candidate_populations. either do so consistently or don't change here

File: pybalance/utils/tests/test_balance_calculators.py [hunk 3]
```diff
@@ -241,3 +241,49 @@ def test_gamma_is_area_between_cdfs():
...
     assert np.isclose(distance_popmat, distance_direct, atol=1e-3)
+
+
+def test_target_subsets():
+
+    matching_data = generate_toy_dataset()
+    target, pool = split_target_pool(matching_data)
+    gamma = BalanceCalculator(matching_data, "gamma", n_bins=20)
+
+    d1 = gamma.distance(pool)
```
Comment (a-hartens) [inline pybalance/utils/tests/test_balance_calculators.py:252]: ummm so you've removed the global scope self.target... so i'm unclear what distance does if there is no target? i.e. what happens when target_subsets = None?

File: pybalance/utils/balance_calculators.py [hunk 4]
```diff
@@ -629,25 +648,25 @@ def _to_array(self, candidate_populations):
...
 
-    def distance(self, candidate_populations):
+    def distance(self, pool_subsets, target_subsets=None):
         # If the user passes a pandas DataFrame, the DataFrame is assumed to
         # represent feature data for only one population and no need for
         # batching; just pass on to the base class. Otherwise, assume
         # candidate_populations refers to patient indices.
-        if isinstance(candidate_populations, pd.DataFrame):
-            return self.balance_calculator.distance(candidate_populations)
+        if isinstance(pool_subsets, pd.DataFrame):
```
Comment (a-hartens) [inline pybalance/utils/balance_calculators.py:656]: you're not checking for type of target_subsets? assumed to be the same type as pool_subsets?

File: pybalance/utils/balance_calculators.py [hunk 5]
```diff
@@ -629,25 +648,25 @@ def _to_array(self, candidate_populations):
...
+            return self.balance_calculator.distance(pool_subsets, target_subsets)
 
         # If the user passes a list, convert it to the underlying backend array
         # type for further processing.
-        if isinstance(candidate_populations, list):
+        if isinstance(pool_subsets, list):
```
Comment (a-hartens) [inline pybalance/utils/balance_calculators.py:661]: target subsets not assumed to require change from list to array?

**Review by a-hartens — COMMENTED** [review]
just type checking of target_subsets otherwise looks good

File: pybalance/utils/balance_calculators.py (hunk 4 above)
Comment (sprivite) [reply pybalance/utils/balance_calculators.py]: fixed

File: pybalance/utils/balance_calculators.py (hunk 5 above)
Comment (sprivite) [reply pybalance/utils/balance_calculators.py:663]: fixed

**Pull request merged**